src/Thumbnail/LiipImagineThumbnail.php

```diff
@@ -11,40 +11,64 @@
 
 namespace Sonata\MediaBundle\Thumbnail;
 
+use Liip\ImagineBundle\Imagine\Cache\CacheManager;
 use Sonata\MediaBundle\Model\MediaInterface;
 use Sonata\MediaBundle\Provider\MediaProviderInterface;
 use Symfony\Component\Routing\RouterInterface;
 
 class LiipImagineThumbnail implements ThumbnailInterface
 {
     /**
+     * @deprecated Since version 3.3, will be removed in 4.0.
+     *
      * @var RouterInterface
      */
-    protected $router;
+    private $router;
```

Comment:
This is a BC-break

Reply:
Keep the property protected please

Reply:
![sonata-project_sonatamediabundle_pull_1393_pullrequestreview-97404967](https://user-images.githubusercontent.com/1597143/36586801-4ba8041e-1883-11e8-9781-5055fe05454d.PNG)
![sonata-project_sonatamediabundle_pull_1393_pullrequestreview-98846546](https://user-images.githubusercontent.com/1597143/36586808-50c41226-1883-11e8-959d-81912427e6aa.PNG)

Hope this will PR will have an end :smile:.

Reply:
You're almost there!

Reply:
Oh ok I didn't realise what this was, the first diff made me think that this was a new property, because it was all green, but it isn't, which means we should revert this change

Reply:
Ah that's because you removed it at first, and then added it back, I understand what happened now.

Reply:
`protected` restored